src/main/scala/co/teapot/tempest/server/TempestDatabaseClient.scala

```diff
@@ -17,64 +17,116 @@ package co.teapot.tempest.server
+
 trait TempestDatabaseClient {
-  /** Returns a map from node id to attribute value for the the given attribute name.  Any nodes
-    * with null attribute value will be omitted from the resulting map.
-    * The implicit anorm.Column[A] parameter just means that A must be some type which Anorm
-    * can extract from a JDBC row. */
-  def getMultiNodeAttribute[A](graphName: String, nodeIds: Seq[Long], attributeName: String)
-                              (implicit c : anorm.Column[A]): collection.Map[Long, A]  // TODO: Delete method
-
-  def getMultiNodeAttributeAsJSON(graphName: String, nodeIds: Seq[Long], attributeName: String): collection.Map[Long, String]
-
-  def getNodeAttribute[A](graphName: String, nodeId: Long, attributeName: String)
-                         (implicit c : anorm.Column[A]): Option[A] =
-    getMultiNodeAttribute(graphName, Seq(nodeId), attributeName).get(nodeId)
-
-  def getNodeAttributeAsJSON(graphName: String, nodeId: Long, attributeName: String): String =
-    getMultiNodeAttributeAsJSON(graphName, Seq(nodeId), attributeName).getOrElse(nodeId, "null")
-
-  /*  If we create a mock database, we might need to go back to these methods:
-    def getNodeStringAttribute(nodeId: Long, attributeName: String): String =
-    getNodeAttribute(nodeId, attributeName, SqlParser.str(1))
-  def getNodeLongAttribute(nodeId: Long, attributeName: String): Long =
-    getNodeAttribute(nodeId, attributeName, SqlParser.long(1))
-  def getNodeBooleanAttribute(nodeId: Long, attributeName: String): Boolean =
-    getNodeAttribute(nodeId, attributeName, SqlParser.bool(1))
-   */
-  def setNodeAttribute(graphName: String,
-                       nodeId: Long,
+
+  def nodeToIntNode(node: Node): IntNode = {
+    nodeToIntNodeOption(node).getOrElse(
+      throw new InvalidNodeIdException(s"No node in ${node.`type`} has id ${node.id}")
+    )
+  }
+
+  def nodeToTempestId(node: Node): Int =
+    nodeToIntNode(node).tempestId
+
+  def nodeToIntNodeOption(node: Node): Option[IntNode] = {
+    val tempestIdOption = getTempestIdsWithAttributeValue(node.`type`, "id", node.id).headOption
+    tempestIdOption map { i => IntNode(node.`type`, i) }
+  }
+
+  def nodeToIntNodeMap(sourceNodes: Iterable[Node]): collection.Map[Node, IntNode] = {
+    val result = new mutable.AnyRefMap[Node, IntNode]()
+    val nodesByType = sourceNodes groupBy { node => node.`type` }
+    for ((nodeType, nodes) <- nodesByType) {
+      val ids = nodes map (_.id)
+      val idToTempestIdString = getSingleTypeNodeAttributeAsJSON(nodeType, ids.toSeq, "tempest_id")
+      for (node <- nodes) {
+        val tempestId = idToTempestIdString.getOrElse(node.id,
+                                                      throw new InvalidNodeIdException(s"Invalid node $node")
+        ).toInt
+        result(node) = IntNode(nodeType, tempestId)
+      }
+    }
+    result
+  }
+
+
+  def tempestIdToNode(nodeType: String, tempestId: Int): Node = {
```

Comment:
This is really like `Node.new(...)`, but we can't use a constructor because `Node` is autogenerated. Right?

Instead, how about call this `CreateNode` (or `CreateThriftNode`)?